titan-compiler/coder.lua

```diff
@@ -1452,33 +1539,41 @@ generate_exp = function(exp, ctx)
         error("not implemented")
 
     elseif tag == ast.Exp.Var then
-        local cstats, lvalue = generate_var(exp.var, ctx)
-        local cvalue
+        local exp_cstats, lvalue = generate_var(exp.var, ctx)
+
         if     lvalue._tag == coder.Lvalue.CVar then
-            cvalue = lvalue.varname
+            return exp_cstats, lvalue.varname
+
         elseif lvalue._tag == coder.Lvalue.SafeSlot then
-            cvalue = get_slot(exp.var._type, lvalue.slot_address)
+            local cvar = ctx:new_cvar(ctype(exp._type))
+            local cstats = util.render([[
+                ${EXP_STATS}
+                ${VAR_DECL} = ${GET_SLOT};
+            ]], {
+                EXP_STATS = exp_cstats,
+                VAR_DECL = c_declaration(cvar),
+                GET_SLOT = get_slot(exp.var._type, lvalue.slot_address),
+            })
+            return cstats, cvar.name
```

Comment:
Since we are returning here we could remove the return at line 1580.

Reply:
I wish this comment had been attached to line 1580 :)

Reply:
GIthub wouldn't allow me.